File: Lectures/06-Inheritance-Polymorphism/start.cc
#include <vector>
#include <iostream>
using namespace std;
#include "ticket.hh"
#include "studentticket.hh"

int main(void) {
  vector<Ticket*> tick;
  tick.push_back(new Ticket());
  tick.push_back(new StudentTicket());

  for (auto t : tick) { cout << t->findBestSeat() << endl; }    
  for (auto t : tick) { cout << t->toString() << endl; }
}

File: Lectures/06-Inheritance-Polymorphism/Start.java
import java.util.ArrayList;
public class Start {
  public static void main(String [] args) {
    ArrayList<Ticket> tick = new ArrayList<>();
    tick.add(new Ticket());
    tick.add(new StudentTicket());

    for(Ticket t : tick) { System.out.println(t.findBestSeat()); }
  }
}